pyteal/ast/abi/address.py

```diff
@@ -1,33 +1,101 @@
+from typing import Union, Sequence, TypeVar
+from collections.abc import Sequence as CollectionSequence
+
+from pyteal.errors import TealInputError
+
+from pyteal.ast.bytes import Bytes
+from pyteal.ast.addr import Addr
+from pyteal.ast.abi.type import ComputedValue, BaseType
 from pyteal.ast.abi.array_static import StaticArray, StaticArrayTypeSpec
 from pyteal.ast.abi.uint import ByteTypeSpec
 from pyteal.ast.expr import Expr
 
-ADDRESS_LENGTH = 32
+ADDRESS_LENGTH_STR = 58
+ADDRESS_LENGTH_BYTES = 32
+
+T = TypeVar("T", bound=BaseType)
+N = TypeVar("N", bound=int)
 
 
 class AddressTypeSpec(StaticArrayTypeSpec):
     def __init__(self) -> None:
-        super().__init__(ByteTypeSpec(), ADDRESS_LENGTH)
+        super().__init__(ByteTypeSpec(), ADDRESS_LENGTH_BYTES)
 
     def new_instance(self) -> "Address":
         return Address()
 
     def __str__(self) -> str:
         return "address"
 
+    def __eq__(self, other: object) -> bool:
+        return isinstance(other, AddressTypeSpec)
+
 
 AddressTypeSpec.__module__ = "pyteal"
 
 
 class Address(StaticArray):
```

Comment:
No need for `TypeVar`s `T` and `N` in this file. Instead, you should make `Address` inherit from `StaticArray[Byte, Literal[32]]` (or replace 32 with `ADDRESS_LENGTH_BYTES`).

Then all instances of `T` and `N` in the rest of the file can be replaced with `Byte` and `Literal[32]`, respectively.

Same thing applies to `string.py`